repo/blob/azure/azure_storage_test.go

```diff
@@ -270,3 +274,134 @@ func TestAzureStorageInvalidCreds(t *testing.T) {
 		t.Errorf("unexpected success connecting to Azure blob storage, wanted error")
 	}
 }
+
+// TestAzureStorageImmutabilityProtection runs through the behavior of Azure immutability protection.
+func TestAzureStorageImmutabilityProtection(t *testing.T) {
+	t.Parallel()
+	testutil.ProviderTest(t)
+
+	// must be with ImmutableStorage with Versioning enabled
+	container := getEnvOrSkip(t, testContainerEnv)
```

Comment:
@KastenMike don't we need to setup the container here with immutability ?

Reply:
it should already be prepared beforehand